### PR TITLE
Fix function resolution with ambiguous coercions

### PR DESCRIPTION
**Pull request by raghavsethi**
This code will be exercised by the upcoming narrowing coercions PR.

### CODE REVIEW

**Comment by martint** [comment]
looks good

**Comment by raghavsethi** [comment]
The last version did not work with type-only coercions. Please review again or suggest alternatives.

File: presto-main/src/main/java/com/facebook/presto/metadata/FunctionRegistry.java [hunk 1]
```diff
@@ -508,33 +508,46 @@ public Signature resolveFunction(QualifiedName name, List<TypeSignature> paramet
         }
 
         // search for coerced matches
-        List<Signature> coercedMatches = new ArrayList<>();
+        List<SqlFunction> coercedCandidates = new ArrayList<>();
+        Signature firstCoercedMatch = null;
```
Comment (martint) [inline presto-main/src/main/java/com/facebook/presto/metadata/FunctionRegistry.java]: Isn't this the first element of the `coercedCandidates` list? Why do we need a separate variable?

**Comment by martint** [comment]
looks good

**Pull request closed without merging**